### ISSUE TITLE
Deploying a blocking function resets "Additional provider token credentials" selections

### ISSUE THREAD

**Issue by samschurter** [firebase/firebase-functions#1484]
### Related issues

None that I could find

### [REQUIRED] Version info

**node:** v20.9.0

**firebase-functions:** 4.5.0

**firebase-tools:** 12.8.1

**firebase-admin:** n/a

### [REQUIRED] Test case

![image](https://github.com/firebase/firebase-functions/assets/7039470/5a02eba3-19d6-4023-a22c-e88bfebd6307)

```
firebase deploy --only functions
```
```
{
  "$schema": "https://raw.githubusercontent.com/firebase/firebase-tools/master/schema/firebase-config.json",
  "functions": {
    "runtime": "nodejs20",
    "source": "./"
  }
}
```

### [REQUIRED] Steps to reproduce

1. Deploy a blocking function
2. set all the correct parameters to get OAuth tokens from the Microsoft provider, per the documentation
    3. use `signInWithPopup()` and request the `offline_access` scope
    4. check the boxes in blocking function configuration to pass the tokens to the blocking function
5. deploy an update to the blocking function that attempts to access the oauth tokens per the documentation
6. function fails because `credential` is `null`.
7. check the configuration in the console, all the checkboxes to include the id/access/refresh tokens have been reset

### [REQUIRED] Expected behavior

- Automatic deploys should NOT break the app by changing settings that can ONLY be configured manually in the console. 
- The console should update the checkbox status when switching between tabs on the Authentication dashboard

### [REQUIRED] Actual behavior

- Automatic deploys DO break the app by changing settings that can ONLY be configured manually in the console. 
- The status of checkboxes on the console should reflect the real state of the setting

### Were you able to successfully deploy your functions?
What is the meaning of success? Are the functions deployed in a usable state? No. Were there any error messages in the deploy process to indicate that something was wrong? Also no.

**Comment by colerogers** [comment]
Hi @samschurter thanks for opening your first issue in Firebase! So the behavior you are describing is actually working as intended. We treat the functions source as our source-of-truth and assume config/settings should come from where you define functions. This is pretty standard on most CI/CD workflows. In order to set those Blocking Function token settings, just attach them directly in your source, like this:
```
export const beforecreate = v2.identity.beforeUserCreated(
  {
    idToken: true,
    accessToken: true,
    refreshToken: true,
  },
  (event) => {
    console.log(event);
});

export const beforesignin = v2.identity.beforeUserSignedIn(
  {
    idToken: true,
    accessToken: true,
    refreshToken: true,
  },
  (event) => {
    console.log(event);
});
```
Now on updates/redeploys, the CLI knows what the blocking config should look like. Cheers!

**Comment by samschurter** [comment]
Thank you for your response and explanation. That makes sense and I agree that's exactly how I would want it to behave.

However, I did struggle in locating this information. Most search results lead to [this page](https://firebase.google.com/docs/auth/extend-with-blocking-functions) or [this page](https://cloud.google.com/identity-platform/docs/blocking-functions) depending on the search query.

These pages, which are nearly identical, unequivocally instruct to "first select the checkbox in the Triggers section of the Include token credentials dropdown menu in the Google Cloud console." Unfortunately, they do not suggest that those options can be set in the code.

![image](https://github.com/firebase/firebase-functions/assets/7039470/f3d062a4-d281-47da-8a6a-bc6f23faa3b7)

There is also no mention on these pages of function options or examples of using the `beforeUserSignIn/beforeUserCreated` functions with an options object.

I found this process pretty frustrating to debug. It could be significantly improved by either expanding the guides or optimizing the API reference for search engine visibility.
